documentation/api/io.strimzi.api.kafka.model.connect.build.Build.adoc

```diff
@@ -0,0 +1,191 @@
+
+[source,yaml,subs=attributes+,options="nowrap"]
+.Example `output` configuration
+----
+apiVersion: {KafkaConnectApiVersion}
+kind: KafkaConnect
+metadata:
+  name: my-connect-cluster
+spec:
+  #...
+  build:
+    output:
+      type: docker # <1>
+      image: my-registry.io/my-org/my-connect-cluster:latest # <2>
+      pushSecret: my-registry-credentials # <3>
+  #...
+----
+<1> The type of the output which should be used by Strimzi. (Required)
+<2> The full name of the image which will be used including the repository and tag. (Required)
+<3> Name of the secret with the container registry credentials. (Optional)
+
+.Using OpenShift ImageStream
+
+Alternatively, you can also use OpenShift ImageStream to store the new container image.
+The ImageStream has to be created manually before deploying the Kafka Connect.
+To use OpenShift ImageStream, you have to set the `type` to `imagestream`, and the `image` field should contain the name of the ImageStream and the tag which should be used.
+For example `my-connect-image-stream:latest`.
+
+[source,yaml,subs=attributes+,options="nowrap"]
+.Example `output` configuration
+----
+apiVersion: {KafkaConnectApiVersion}
+kind: KafkaConnect
+metadata:
+  name: my-connect-cluster
+spec:
+  #...
+  build:
+    output:
+      type: imagestream # <1>
+      imageStream: image-stream-name # <2>
+  #...
+----
+<1> The type of the output which should be used by Strimzi. (Required)
+<2> The name of the ImageStream and tag. (Required)
+
+=== `plugins`
+
+The connector plugins which will be added to the container image are configured in the `.spec.build.plugins` section of the `KafkaConnect` custom resource.
+Each connector plugin needs to have a name which is unique within the Kafka Connect deployment.
+Additionally, it contains a list of artifacts which it consists of.
+These artifacts will be downloaded by Strimzi, added to the new container image, and used in the Kafka Connect deployment.
+The connector plugin artifacts can also include additional components such as (de)serializers and so on.
+Each connector plugin will be downloaded into a separate directory so that the different connectors and their dependencies are properly sand-boxed.
+The `plugins` section is required.
+Each plugin has to be configured with at least one `artifact`.
+
+[source,yaml,subs=attributes+,options="nowrap"]
+.Example `plugins` configuration with two connector plugins
+----
+apiVersion: {KafkaConnectApiVersion}
+kind: KafkaConnect
+metadata:
+  name: my-connect-cluster
+spec:
+  #...
+  build:
+    output:
+      #...
+    plugins: # <1>
+      - name: debezium-postgres-connector
+        artifacts:
+          - type: tgz
+            url: https://repo1.maven.org/maven2/io/debezium/debezium-connector-postgres/1.3.1.Final/debezium-connector-postgres-1.3.1.Final-plugin.tar.gz
+            sha512sum: 962a12151bdf9a5a30627eebac739955a4fd95a08d373b86bdcea2b4d0c27dd6e1edd5cb548045e115e33a9e69b1b2a352bee24df035a0447cb820077af00c03
```

Comment:
Can we snip these hashes?

Reply:
I was thinking about it. I think we have two options ... either use some completely artificial examples where we can keep some short URL and short (invalid) checksum. Or we can use areal connectors as we have there now. I do not think we should mix it - have a real URL with wrong checksum since that would confuse people who try to use it. 

I thought about the length already when writing it. But it seemed ok to me to go with the real thing. But I do not have strong opinions. So if you think I can change it to fake values (which I used anyway for the JAR artifact where I did not have a nice example - apart from my own connector which a bit seemed weird to use here).

Reply:
I see. No strong opinions. It was really about the presentation, which is secondary if you think real examples might even be used. One potential issue on using real examples is maintenance, and how long we expect them to be relevant.